### PR TITLE
Add new Maven build for Scala package

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##
This PR is a follow up with @frankfliu PR to add maven run configurations. We also enabled MKLDNN unit test run on Scala
@frankfliu @marcoabreu @zachgk @andrewfayres @piyushghai 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by anirudhacharya** [comment]
@mxnet-label-bot add [pr-awaiting-review, Scala]

File: ci/docker/runtime_functions.sh [hunk 1]
```diff
@@ -847,23 +847,23 @@ unittest_ubuntu_python3_quantization_gpu() {
 unittest_ubuntu_cpu_scala() {
     set -ex
     scala_prepare
-    make scalapkg USE_BLAS=openblas USE_DIST_KVSTORE=1 ENABLE_TESTCOVERAGE=1 CI=1
-    make scalaunittest USE_BLAS=openblas USE_DIST_KVSTORE=1 ENABLE_TESTCOVERAGE=1 CI=1
+    cd scala-package
+    mvn -B install
```
Comment (zachgk) [inline ci/docker/runtime_functions.sh]: This should be integration-test (and next two)

File: ci/jenkins/Jenkins_steps.groovy [hunk 2]
```diff
@@ -825,7 +825,21 @@ def test_unix_scala_cpu() {
       node(NODE_LINUX_CPU) {
         ws('workspace/ut-scala-cpu') {
           timeout(time: max_time, unit: 'MINUTES') {
-            utils.unpack_and_init('cpu', mx_dist_lib, true)
+            utils.unpack_and_init('cpu', mx_lib, true)
```
Comment (zachgk) [inline ci/jenkins/Jenkins_steps.groovy]: I don't think we need mx_dist_lib anymore. If so, we should just delete the variable

File: scala-package/spark/README.md [hunk 3]
```diff
@@ -16,7 +16,8 @@ Checkout the [Installation Guide](http://mxnet.io/get_started/setup.html) contai
 Compile the Scala Package by
 
 ```bash
-make scalapkg
+cd scala-package
+mvn install
```
Comment (zachgk) [inline scala-package/spark/README.md]: The equivalent of make scalapkg is "mvn package"

File: tests/jenkins/run_test_amzn_linux_gpu.sh [hunk 4]
```diff
@@ -65,5 +65,5 @@ nosetests3 --verbose tests/python/train
 #julia -e 'try Pkg.clone("MXNet"); catch end; Pkg.checkout("MXNet"); Pkg.build("MXNet"); Pkg.test("MXNet")' || exit -1
 
 echo "BUILD scala_test"
-make scalapkg
-make scalatest
+cd scala-package
+mvn install
```
Comment (zachgk) [inline tests/jenkins/run_test_amzn_linux_gpu.sh]: Shouldn't this and below be "integration-test" or "integration-test -Dskip.test=false"?

File: contrib/clojure-package/README.md [hunk 5]
```diff
@@ -193,8 +193,8 @@ The outcome of this step will be a shared library `lib/libmxnet.so` that is used
 - Build and install the Scala package in your local Maven directory using the following commands:
 
   ```bash
-  make scalapkg
-  make scalainstall
+  cd scala-package
+  mvn install
```
Comment (piyushghai) [inline contrib/clojure-package/README.md]: I would rather make is : ```mvn clean install```

File: scala-package/spark/README.md [hunk 6]
```diff
@@ -17,7 +17,7 @@ Compile the Scala Package by
 
 ```bash
 cd scala-package
-mvn install
+mvn package
```
Comment (piyushghai) [inline scala-package/spark/README.md]: Why change it from install to package ?

File: contrib/clojure-package/README.md (hunk 5 above)
Comment (lanking520) [reply contrib/clojure-package/README.md]: As a new user tutorial, I think it should be fine to just do install

File: scala-package/spark/README.md (hunk 6 above)
Comment (lanking520) [reply scala-package/spark/README.md]: @zachgk

File: tests/jenkins/run_test_amzn_linux_gpu.sh [hunk 7]
```diff
@@ -65,5 +65,5 @@ nosetests3 --verbose tests/python/train
 #julia -e 'try Pkg.clone("MXNet"); catch end; Pkg.checkout("MXNet"); Pkg.build("MXNet"); Pkg.test("MXNet")' || exit -1
 
 echo "BUILD scala_test"
-make scalapkg
-make scalatest
+cd scala-package
+mvn integration-test -Dskip.test=false
```
Comment (zachgk) [inline tests/jenkins/run_test_amzn_linux_gpu.sh]: You should use -DskipTests=false. Also, why is this one "make scalatest -> integration tests" while run_test_ubuntu.sh is "make scalatest -> unit tests"?

Comment (lanking520) [reply tests/jenkins/run_test_amzn_linux_gpu.sh]: Since this is tested on GPU, we should enable the integration test there

**Pull request merged**